plugins/win-capture/window-capture.c

```diff
@@ -234,6 +288,29 @@ static void wc_tick(void *data, float seconds)
 		dc_capture_free(&wc->capture);
 		dc_capture_init(&wc->capture, 0, 0, rect.right, rect.bottom,
 				wc->cursor, wc->compatibility);
+
+		struct dstr title = {0};
+		struct dstr class = {0};
+		struct dstr exec = {0};
+		struct dstr win = {0};
+
+		get_window_exe(&exec, wc->window);
+		get_window_class(&class, wc->window);
+		get_window_title(&title, wc->window);
+
+		dstr_printf(&win, "%s:%s:%s", title.array, class.array,
+			    exec.array);
+
+		obs_data_t *settings = obs_source_get_settings(wc->source);
+
+		obs_data_set_string(settings, "window", win.array);
```

Comment:
Shouldn't `"window"` already be set at this point? Without that, there would be no setting at all, and window capture wouldn't capture anything iirc?

Reply:
This changes window to reflect window title changes as they happen. This bit was added because without it, the change callback would check if any dropdown item matched, and wouldn't find a match if the window title was different, resulting in the creation of a disabled item, even though it was capturing a window. Thus it failed my test case for:

> I also made sure that when the window title changed, that the dropdown still accurately reflects the window title when you open up properties.

I realize though that having the title to match against change also has problems, so I'll probably change it to use a different key that the callback responds to.

Reply:
I have updated the test case wording to be more clear:
> I also made sure that when the window title changed, that the dropdown accurately reflects the window title of the currently captured window when you open up properties.